src/System.Management.Automation/engine/Modules/ModuleUtils.cs

```diff
@@ -43,22 +54,25 @@ internal static bool IsPossibleModuleDirectory(string dir)
                 }
                 catch { }
             }
-#endif
 
             return true;
         }
 
         /// <summary>
-        /// Get a list of all module files
-        /// which can be imported just by specifying a non rooted file name of the module
-        /// (Import-Module foo\bar.psm1;  but not Import-Module .\foo\bar.psm1)
+        /// Get all module files by searching the given directory recursively.
+        /// All sub-directories that could be a module folder will be searched.
         /// </summary>
-        /// <remarks>When obtaining all module files we return all possible
-        /// combinations for a given file. For example, for foo we return both
-        /// foo.psd1 and foo.psm1 if found. Get-Module will create the module
-        /// info only for the first one</remarks>
+        /// <remarks>
+        /// When obtaining all module files we return all possible combinations for a given file.
+        /// For example, for 'foo' we return both 'foo.psd1' and 'foo.psm1' if found.
+        /// Get-Module will create the module info only for the first one.
+        /// </remarks>
         internal static IEnumerable<string> GetAllAvailableModuleFiles(string topDirectoryToCheck)
         {
+            if (!Directory.Exists(topDirectoryToCheck)) { yield break; }
+            var options = Utils.PathIsUnc(topDirectoryToCheck) ? s_uncPathEnumOptions : s_defaultEnumOptions;
```

Comment:
Would it be possible to use `EnumerationOptions` instead of `var` here?

Reply:
Sure, fixed.